### ISSUE TITLE
bug: step definitions never get's reloaded

### ISSUE THREAD

**Issue by NoNameProvided** [TheBrainFamily/cypress-cucumber-preprocessor#1]
When using `npx cypress open` to write tests, it would be great to have auto-updated step definitions. Now they are loaded only once on start only so I need to re-run cypress every time what is very slow.

**Comment by eaustin** [comment]
I believe have the same problem.

**Comment by NoNameProvided** [comment]
Here is the gist: https://gist.github.com/NoNameProvided/14ba8dee2529de258804454d2c3c99c8

**Comment by lgandecki** [comment]
He guys, github didn't automatically make me watch my own repo, so I missed those comments! Watching it now and will be more responsive.

Yes, I wanted to have something out of the door because I needed it for one of my projects as well. Reloading after changes is not optimal, but it worked from the client perspective and allowed me to use cypress while having cucumber-like specs.

@NoNameProvided Any particular reason you decided to create a new repo and github gists instead of making your changes in a fork? I would really appreciate that - then I can easily compare the changes, and merge them in. That's the github/open source spirit! :-) And the usual way of doing things.. 

I'm more than happy to make you a contributor with write rights here, if you make some improvements.

**Comment by lgandecki** [comment]
At this moment the only way to incorporate your changes is such that they will look like my commits, not yours, so please make a fork in the next few days, so the credit to your work can be visible in the project. 
Thanks ! :)

As for reloading when step_definitions change - that's tricky, because cypress doesn't know about them till the run time. Having said that, we could use the same AST parsing mechanism to detect which files will be needed by a given .feature file, and watch them accordingly. A bit of duplication of logic but won't hurt, and the usability will be much nicer. If you want to work on that we can pair up, or I can explain you more what needs to be done. Otherwise I might have time in the next two weeks to make it work the way you would expect

**Comment by NoNameProvided** [comment]
> Any particular reason you decided to create a new repo and github gists instead of making your changes in a fork

I have created a new repo because you weren't responsive and we write in our code in TypeScript anyway. 

> That's the github/open source spirit! :-) And the usual way of doing things..

Yes, that's the reason why I contribute to various open-source projects, but sometimes it's just easier to take your own way.

**Comment by fcurella** [comment]
Hi,

I've got this working with features and step definitions reload. Should I send a PR to NoNameProvided/cypress-cucumber-preprocessor or here?

**Comment by lgandecki** [comment]
@fcurella Please send it here. The other repo is empty, the guy also doesn't have access to the npm name cypress-cucumber-preprocessor. He's made two tiny changes while I wasn't responsive for ~10 days! I put a lot of work into this, and don't see a reason why this should be moved somewhere else and why I should lose control? I've been doing OpenSource for 10 years now and I haven't seen a situation like this to be honest :)

@NoNameProvided Even if someone is not responsive it is a good taste to make a fork! Instead of just creating a new repo, which doesn't give credit to the original author, makes the collaboration harder - by inability to send patches and compare the code. 
You said "I made the *.feature files reload and I have a few other improvements as well," but in reality you've made exactly ONE other "improvement" 

```javascript
  describe(gherkinAst.feature.name, () => {
    gherkinAst.feature.children.forEach(createTestFromScenario);
  });
```

instead of
```  
gherkinAst.feature.children.forEach(createTestFromScenario);
```

which makes it a bit dishonest to be fair (just say "I made one tiny change", if that's all you did), and you made it hard to compare/hard for people realize this because you haven't forked.. adding 10 days of inactivity (and you not trying to @lgandecki me here, or ping me or twitter), doesn't add up to a good reason to pretend you've done this work from scratch by yourself on a separate repo :). Putting this as a gist without package.json that mentions my name, or at least adding a comment about where the original code is from - that's really not appreciated.
Also, you say you write code in typescript - fine, but the gists you sent aren't typescript. If you wrote the whole thing from scratch, in TypeScript, basing it on sam ideas I've had here - fine. I'd still expect a CREDIT in your readme though :)

Just place nice!

**Comment by NoNameProvided** [comment]
@lgandecki 
> He's made two tiny changes while I wasn't responsive for ~10 days! And even less then that (6 days) between his comment and the decision to make a gist instead of a fork.

> which makes it a bit dishonest to be fair (just say "I made one tiny change", if that's all you did), and you made it hard to compare/hard for people realize this because you haven't forked..

Chill a little, this tone won't help. I haven't made a fork because I use a different language than you do and I have decided to implement everything from scratch anyway.

Currently, now it's totally diverged as here is one file hold everything for us it's structured into files, a few more features have been added also. So yes I will make a different repo because I don't want to wait or argue about what to get merged and what not when I have a clear view of what we need to use.

@fcurella Thanks, I see you opened a pull request at my repo, thank you for that! Sadly I cannot merge it now as our code looks totally different internally. However, I will let you know when we publish it finally so you can check it and use it if you see it fit.

**Comment by lgandecki** [comment]
won't help with what? :)

you took my code, put it as a gist without any credit to my work (that's a violation of the MIT license, btw), made a tiny change to it but pretended you've done much more, and since you didn't fork you made it hard to compare and see that. Just because some piece of code is on github doesn't mean you can do whatever you want with it, and tell people to chill when being called out. :) 

My first reply to you was nice and inviting to cooperation, I put a blind eye to above mentioned facts. I honestly expected you to play nice at that point. @fcurella made some solid work here, and somehow didn't think it would be better to create his own repo.

**Comment by lgandecki** [comment]
@fcurella I'd love to add your contribution here and give you the credit deserved, would you be so kind to fork this one, paste your changes over the files in this repo and create another PR? I've just tried to use your fork but since it has a different commit history, github won't allow me to compare/merge this properly :(

**Comment by fcurella** [comment]
@lgandecki sorry about the confusion, I had the impression that you and @NoNameProvided were working together. I'll close the original PR and submit it here.

**Comment by NoNameProvided** [comment]
> won't help with what? :)

You are feeling offended as I can see and your tone reflects that, but I don't get it why. 

> you took my code, put it as a gist without any credit to my work (that's a violation of the MIT license, btw)

First, do you get it that it is a secret gist? So no one can see it just who visits it through this links posted here in your repo. You act like I am getting really famous now because of that gist without crediting you anything. Just to be fair I will add your license so if I would get a billion $ offer from that code you will be the one they will go instead now.

Btw if I follow your logic I would not be allowed to paste code-snippets in your issues here without including the license. Do you do that all the time in other repos? I don't think so, why you expect it from me then? This feels like storm in a teacup for me.

> made a tiny change to it but pretended you've done much more

I didn't pretend I did much more, where have you read that? I stated I will do more, and indeed I will.

> My first reply to you was nice and inviting to cooperation, I put a blind eye to above mentioned facts. 

Yes, and I refused because it fits me better. Then you went wild and felt insulted. Because you wrote a pice of code it doesn't means everybody needs to use it or contribute to it.

> and somehow didn't think it would be better to create his own repo.

Yes, because it fits his/her needs better.

I hope we are good now.

**Comment by lgandecki** [comment]
I removed the out of the topic discussion. You don't have to use this NoName, you are right. I never asked you. I only ask you to respect my work and don't steal it or bash it with no arguments (happy to see how you made things better, when you finally decide to enlighten us).

@fcurella thanks, I will review this during the next 24 hrs :-) I already looked briefly but I want to test and understand exactly what you've changed and how it works first.

**Comment by cicely-f** [comment]
I'm running Cypress 3.0.1 and the latest version of this plugin. The step definitions are still not reloaded, and I am forced to close and restart Cypress to get the updated steps to run.

To reproduce:
1. write a feature file with several Given, When, Then steps
2. write corresponding step definition, implementing only the Given
3. see error message about missing step implementation when running Cypress
4. implement the When step
5. watch Cypress re-run the apparently previous version of the step definition, since it returns the same error message as in step 3
6. close and reopen Cypress
7. see that the error message changes to missing implementation of the Then step
